src/main/java/org/jvnet/hudson/test/RestartableJenkinsRule.java

```diff
@@ -21,34 +20,26 @@
 import java.util.concurrent.Callable;
 import java.util.logging.Level;
 import java.util.logging.Logger;
+import org.junit.After;
 import org.junit.Assert;
 import org.junit.rules.MethodRule;
 import org.junit.runner.Description;
 import org.junit.runners.model.FrameworkMethod;
 import org.junit.runners.model.Statement;
 
 /**
- * Provides a pattern for executing a sequence of steps.
- * In between steps, Jenkins gets restarted.
- *
- * <p>
- * To use this, add this rule instead of {@link JenkinsRule} to the test,
- * then from your test method, call {@link #then} repeatedly.
- * You may test scenarios related to abrupt shutdowns or failures to start using {@link #thenWithHardShutdown} and
- * {@link #thenDoesNotStart}.
- * <p>
- * The rule will evaluate your test method to collect all steps, then execute them in turn and restart Jenkins in
- * between each step. Consider using {@link JenkinsSessionRule} if you want each step to be executed immediately when
- * {@link #then} is called.
- * <p>
- * If your test requires disabling of a plugin then the default {@link PluginManager} ({@link TestPluginManager}) used for tests
- * will need to be changed to {@link UnitTestSupportingPluginManager}.
- * This can be accomplished by annotating the test with {@code @WithPluginManager(UnitTestSupportingPluginManager.class)}.
- * 
  * @author Kohsuke Kawaguchi
- * @see JenkinsRule
  * @since 1.567
+ * @deprecated New code should use {@link JenkinsSessionRule}.
```

Comment:
hmm shouldn't new code use `@WithJenkins` (JUnit 5)? with which you can use `.restart()` directly on the jenkinsrule instance?

Reply:
I have no experience with the JUnit 5 facilities and see no reason to ever switch from 4.

Regarding `JenkinsRule.restart`, I would not recommend it. It seems to make a copy of `$JENKINS_HOME`, which `JenkinsSessionRule` does not need to do. Also it does not support performing test steps in between Jenkins sessions, as `JenkinsSessionRule` does.

Reply:
fyi @strangelookingnerd  ^^

Reply:
I‘m all for deprecating the rule. Overall I still think the transition towards JUnit5 is something we should promote rather than relying on a framework that is no longer maintained. I understand @jglick point that as of now there is little reason to switch, but that may change in the future.

Reply:
> something we should promote

You should add Javadoc to various 4-specific utilities noting their replacements.